packages/react-sdk/src/components/text-block.tsx

```diff
@@ -1,10 +1,19 @@
-import { forwardRef, type ElementRef, type ComponentProps } from "react";
+import {
+  createElement,
+  forwardRef,
+  type ElementRef,
+  type ComponentProps,
+} from "react";
 
 export const defaultTag = "div";
 
-export const TextBlock = forwardRef<
-  ElementRef<typeof defaultTag>,
-  ComponentProps<typeof defaultTag>
->((props, ref) => <div {...props} ref={ref} />);
+// We don't want to enable all tags because Box is usually a container and we have specific components for many tags.
+type Props = ComponentProps<typeof defaultTag> & {
+  tag?: "div" | "span" | "figcaption";
+};
 
-TextBlock.displayName = "TextBlock";
+export const TextBlock = forwardRef<ElementRef<typeof defaultTag>, Props>(
+  ({ tag = defaultTag, ...props }, ref) => {
+    return createElement(tag as string, { ...props, ref });
```

Comment:
Why as string?

Reply:
Seems like box works without it

Reply:
I think it used to not work without it, removing from both